### PR TITLE
If no thumbnail exists, check if resized version exist and use this

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
This is useful if you have a very large image folder, scroll to a decent point and then immediately click on an image. While swiping the resized version is fetched.
If you then browse the file listing, the app would download the thumbnails, but now uses the resized images as source.

Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

### CODE REVIEW

**Comment by tobiasKaminsky** [comment]
@AndyScherzinger ping :-)

**Comment by AndyScherzinger** [comment]
:+1: Nice solution/idea :)

File: src/main/java/com/owncloud/android/datamodel/ThumbnailsCacheManager.java [hunk 1]
```diff
@@ -528,55 +528,62 @@ private Bitmap doThumbnailFromOCFileInBackground() {
...
-                                Log_OC.d(TAG, e.getMessage(), e);
-                            } finally {
-                                if (getMethod != null) {
-                                    getMethod.releaseConnection();
+                    if (resizedImage != null) {
+                        thumbnail = ThumbnailUtils.extractThumbnail(resizedImage, pxW, pxH);
+                    } else {
+                        // Download thumbnail from server
+                        OwnCloudVersion serverOCVersion = AccountUtils.getServerVersion(mAccount);
+                        if (mClient != null && serverOCVersion != null) {
```
Comment (mario) [inline src/main/java/com/owncloud/android/datamodel/ThumbnailsCacheManager.java]: serverOCVersion cannot be null :P

File: src/main/java/com/owncloud/android/datamodel/ThumbnailsCacheManager.java [hunk 2]
```diff
@@ -528,55 +528,62 @@ private Bitmap doThumbnailFromOCFileInBackground() {
...
+                            if (serverOCVersion.supportsRemoteThumbnails()) {
```
Comment (mario) [inline src/main/java/com/owncloud/android/datamodel/ThumbnailsCacheManager.java]: please use AccountUtils.getServerVersion(mAccount).supportsRemoteThumbnails() where possible.

Also put if (mClient != null && AccountUtils.getServerVersion(mAccount).supportsRemoteThumbnails()) in the else above.

File: src/main/java/com/owncloud/android/datamodel/ThumbnailsCacheManager.java (hunk 1 above)
Comment (tobiasKaminsky) [reply src/main/java/com/owncloud/android/datamodel/ThumbnailsCacheManager.java]: #2211 is not yet merged and there it is already changed

File: src/main/java/com/owncloud/android/datamodel/ThumbnailsCacheManager.java (hunk 2 above)
Comment (tobiasKaminsky) [reply src/main/java/com/owncloud/android/datamodel/ThumbnailsCacheManager.java]: The first one is already done in #2211.
I'll wait until #2211 is merged, rebase this and the adjust the second one.

**Comment by tobiasKaminsky** [comment]
@mario rebased & implemented your idea :

**Comment by mario** [comment]
👍

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky all green except this _continuous-integration/drone/push_ check. This started within the last week or so I think. Any idea where this comes from?

**Comment by MorrisJobke** [comment]
> @tobiasKaminsky all green except this continuous-integration/drone/push check. This started within the last week or so I think. Any idea where this comes from?

Yes - somebody configured it to be required (and thus GitHub automatically adds it (you can see this by the missing details link - this link gets added once the service actually says, that it received the check). As push checks are not executed by default by drone it is not filled and therefore missing ;)

bildschirmfoto 2018-03-20 um 07 56 19

**Comment by tobiasKaminsky** [comment]
Thanks @MorrisJobke for your help, I changed the settings back to the previous one
- only PR
- no github pull request review is requried

**Pull request merged**